### ISSUE TITLE
Final slash in printer URL not normalized in REST calls

### ISSUE THREAD

**Issue by davidzwa** [OctoFarm/OctoFarm#277]
**Description**
Adding a slash after the printer URL causes the connection to fail (due to // ending up in the OctoPrint REST calls)

**To Reproduce**
1. Go to 'printers'
2. Adjust printer URL to end with `/`
3. Save
4. See error connecting to printer Host 

**Expected behavior**
Final slash should be normalized before saved to database or when doing external API calls.

**Screenshots**
URL with slash failing
![12-feb final slash printer url not normalized](https://user-images.githubusercontent.com/6005355/107773985-b8289780-6d3e-11eb-9a4a-52dafb31e5ec.PNG)
URL without slash working fine
![12-feb without slash printer connects](https://user-images.githubusercontent.com/6005355/107773987-b8c12e00-6d3e-11eb-91b7-15bd4d9929d3.PNG)

**Version**
- Master branch 1.1.10 version
**Additional context**
Did also check development if any fix was underway, could not spot it.

**Comment by davidzwa** [comment]
Can be closed on the spot if this is already fixed.

**Comment by NotExpectedYet** [comment]
![image](https://user-images.githubusercontent.com/61137988/107776134-19e60300-6d39-11eb-8b4e-9ee6aa9b9d78.png)

This was fixed ages ago, how are you producing that? Editing there? that may skip the above as it wouldn't re-setup the defaults.

**Comment by davidzwa** [comment]
Yes continuously editing the URL as I experienced mDNS issues regularly at home (hostnames exist, but not resolving: yay!). 
Also the default came with slash.

I'll check right now on dev, if it persists, just to be sure.

**Comment by NotExpectedYet** [comment]
It's not always DNS aha

The fix is in when you initially save the printer, as I said it may get skipped currently with updating. 

What default came with a slash?

**Comment by davidzwa** [comment]
Double checked on dev.

So, what I filled in as URL initially was `http://prusa/`, and that failed to resolve (btw I have the same problems within PrusaSlicer, it tests fine but won't send prints to OctoFarm afterwards... my knowledge with FQDN is not 100% expert, I admit that. Just had a lot of instability issues with it last year during an IoT internship.)

I guess, don't really recall, changing `http://prusa` which connected to host but failed on websocket.
![image](https://user-images.githubusercontent.com/6005355/107777882-b2818080-6d43-11eb-9ee5-4296967516e7.png)

Then I changed to IP directly (without ending slash), and that worked. Changed to the one with slash, and as you state, we know what goes wrong there.
